splicemachine/spliceengine
# DB-10968/DB-11156 SET_GLOBAL_DATABASE_PROPERTY should check existence of properties, restrict execution of GET_REGION_SERVER_CONFIG_INFO

# Description

- DB-10968 SYSCS_SET_GLOBAL_DATABASE_PROPERTY should check existence of properties
- DB-11156 restrict execution of SYSCS_GET_REGION_SERVER_CONFIG_INFO to admin only

# Motivation DB-11156

SYSCS_GET_REGION_SERVER_CONFIG_INFO shows internal server information. Before this fix, every user could call it, after it, only admins.

# How to test DB-11156

```
-- create a simple (non-admin) user
CALL SYSCS_UTIL.SYSCS_CREATE_USER('fred', 'fredpassword');

-- connect as Fred
connect 'jdbc:splice://localhost:1527/splicedb;user=fred;password=fredpassword';

call SYSCS_UTIL.SYSCS_GET_REGION_SERVER_CONFIG_INFO(null, 0);
-- this should not be possible, as this prints internal server configuration which should not be accessible 
```
# Motivation DB-10968

When calling `SYSCS_SET_GLOBAL_DATABASE_PROPERTY` we don't give feedback if that property is even valid, so e.g. if the user does something like

```
splice> call SYSCS_UTIL.SYSCS_SET_GLOBAL_DATABASE_PROPERTY( 'not.existing.property', true);
Statement executed.
ELAPSED TIME = 991 milliseconds
```

The output doesn't show that we didn't set a meaningful property, so making small errors like using `splice.db2.VARCHAR.compatible` or `db2.varchar.compatible` instead of `splice.db2.varchar.compatible` will not report any warnings.

# How to test DB-10968

new code will report a warning when setting a not-known property:

```
splice> call SYSCS_UTIL.SYSCS_SET_GLOBAL_DATABASE_PROPERTY( 'db2.varchar.compatible', true);
name                |value                                                                                                                                                                               
------------------------------------------------------------------------------
KEY                 |db2.varchar.compatible                                                                                                                                                              
NEW VALUE           |true                                                                                                                                                                                
PREVIOUS VALUE      |NULL                                                                                                                                                                                
                    |                                                                                                                                                                                    
!!! WARNING !!!     |Database Property 'db2.varchar.compatible' seems to be unknown!                                                                                                                     

5 rows selected
```

For correct properties, it will show the previous value:

```
ELAPSED TIME = 31 milliseconds
splice> call SYSCS_UTIL.SYSCS_SET_GLOBAL_DATABASE_PROPERTY( 'splice.db2.varchar.compatible', true);
name                |value                                                                                                                                                                               
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
KEY                 |splice.db2.varchar.compatible                                                                                                                                                       
NEW VALUE           |true                                                                                                                                                                                
PREVIOUS VALUE      |NULL                                                                                                                                                                                

3 rows selected
ELAPSED TIME = 29 milliseconds
```
